src/parts.rs

```diff
@@ -0,0 +1,162 @@
+//! Parts: Split the input to tokens with the common naming conventions.
+
+use std::{borrow::Cow, ops::Deref};
+
+#[derive(Debug, PartialEq)]
+pub struct Parts(Vec<String>);
+
+impl Parts {
+    pub fn to_pascalcase(&self) -> String {
+        self.iter().fold(String::new(), |mut acc, token| {
+            for (idx, c) in token.chars().enumerate() {
+                // We convert the first character of token to uppercase,
+                // then the rest to lowercase. As we've converted all the characters
+                // in token to lowercase (see From::<&str>::from), we just push those
+                // characters to `acc` with no any modification.
+                if idx == 0 {
+                    acc.push(c.to_ascii_uppercase());
+                } else {
+                    acc.push(c);
+                }
+            }
+            acc
+        })
+    }
+
+    pub fn to_camelcase(&self) -> String {
+        self.iter().enumerate().fold(String::new(), |mut acc, (token_idx, token)| {
+            for (c_idx, c) in token.chars().enumerate() {
+                // If the token is not in the first order ("camel" in ["camel", "case"]),
+                // we convert the first character of token to uppercase (["camel", "Case"]), 
+                // then the rest to lowercase. As we've converted all the characters
+                // in token to lowercase (see From::<&str>::from), we just push those
+                // characters to `acc` with no any modification.
+                if token_idx != 0 && c_idx == 0 {
+                    acc.push(c.to_ascii_uppercase());
```

Comment:
this, for example, clearly won't work with non-ascii

Reply:
We can replace it with [`.to_uppercase()`](https://doc.rust-lang.org/stable/std/primitive.char.html#method.to_uppercase), which has Unicode support, if you think it is necessary.